paddle/fluid/platform/device_context.h

```diff
@@ -184,6 +217,18 @@ class CUDADeviceContext : public DeviceContext {
     callback_manager_->Wait();
   }
 
+#if CUDA_VERSION >= 9000
+  /*! \brief CublasCall may need to change cublas's config,
+   *  but the cublas may be hold by multi-thread, so we should
+   *  add lock here. */
+  template <typename Callback>
+  void CublasCall(Callback callback, cublasMath_t new_math) {
+    std::lock_guard<std::mutex> guard(cublas_mtx_);
```

Comment:
Maybe we can just set TensorCoreMath, and fallback to default math. this lock may limited the speed in dense computations.

Reply:
This is a good question.
In fact, I found that when we use `cublasSgemmEx` to attempt to use `Tensor Core` during calling gru_op (the input's size doesn't satisfy the Tensor Core require, so it should fall back to normal `cublasSgemm`), the speed is a bit slower than before. I don't know the further reason, but only know that when gru_op calls GEMM, it specify the ldc( code [here](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/math/gru_compute.cc#L42)). So I think `cublasSgemmEx` may be not equal to `cublasSgemm` when the input's size doesn't satisfy the `Tensor Core` requirement.